backend/src/models/chat.ts

```diff
@@ -39,6 +31,10 @@ const modelConfigIds = [
 
 type MODEL_CONFIG_ID = (typeof modelConfigIds)[number];
 
+type ChatModelConfigurations = {
```

Comment:
There must be a reason you pluralised this, right? Is there a clash with another type, or did you feel it made more sense as a plural?

Reply:
I prefer it that way.

If each of top p, `frequency penalty etc are one configuration, then the set of all of them should be called configurations.

Reply:
That's fine, but then export it as that - currently you're exporting it as ChatModelConfiguration.